### PR TITLE
Have the owner provide the balances for the candidates

### PR DESCRIPTION
**Pull request by DemiMarie**
rather than the balances being hard-coded into genesis.

### CODE REVIEW

**Comment by DemiMarie** [comment]
`npm run test` fails, but it also fails on [`master`](https://github.com/poanetwork/posdao-test-setup/tree/b60994bd16dfdeab39d3cba7a29f7d105a083cd4).

File: scripts/addCandidateBalances.js [hunk 1]
```diff
@@ -12,7 +12,6 @@ async function main() {
   assert(accounts != null && Object.getPrototypeOf(accounts) === Object.prototype);
   for (const candidate of constants.CANDIDATES) {
     assert(!Object.prototype.hasOwnProperty.call(accounts, candidate.staking));
-    accounts[candidate.staking] = { balance: '0x100000000000000000' };
```
Comment (phahulin) [inline scripts/addCandidateBalances.js]: Is this file required at all now?

File: test/0_staking.js [hunk 2]
```diff
@@ -10,22 +11,53 @@ const expect = require('chai')
...
 
+    it('makes the owner the ERC20 to Native Bridge', async () => {
+        const tx = await SnS(web3, {
+            from: OWNER,
+            to: BlockRewardContract.address,
+            method: BlockRewardContract.instance.methods.setErcToNativeBridgesAllowed([OWNER]),
+            gasPrice: '0x0',
+        });
+        console.log(tx);
+        pp.tx(tx);
```
Comment (phahulin) [inline test/0_staking.js]: `pp.tx(tx)` is just a less-detailed pretty-print of `console.log(tx)`, so only one of them should be kept. I'd say, leave pp.tx because it's easier to read.
(Also in `Owner issues initial balances to candidates` test)

File: test/0_staking.js [hunk 3]
```diff
@@ -10,22 +11,53 @@ const expect = require('chai')
...
 describe('Candidates make stakes on themselves', () => {
+    'use strict';
     var minStake;
     var minStakeBN;
     before(async () => {
         minStake = await ValidatorSetContract.instance.methods.getCandidateMinStake().call();
         minStakeBN = new BN(minStake.toString());
     });
 
+    it('makes the owner the ERC20 to Native Bridge', async () => {
```
Comment (phahulin) [inline test/0_staking.js]: I suppose all tests would require sending some native coins to certain accounts. So maybe it's better to make `OWNER => bridge` outside the test suite, right after the ERC20 is deployed?

File: test/0_staking.js [hunk 4]
```diff
@@ -46,34 +78,44 @@ describe('Candidates make stakes on themselves', () => {
...
         let stakeBN = minStakeBN.clone();
         console.log('**** stake = ' + stakeBN.toString());
-        for (candidate of constants.CANDIDATES) {
+        for (const candidate of constants.CANDIDATES) {
             await canStakeAndWithdraw(web3);
             console.log('**** candidate =', JSON.stringify(candidate));
-            let iStake = await ValidatorSetContract.instance.methods.stakeAmount(candidate.staking, candidate.staking).call();
-            let iStakeBN = new BN(iStake.toString());
-            let tx = await SnS(web3, {
+            const iStake = await SnS(web3, {
```
Comment (phahulin) [inline test/0_staking.js]: `stakeAmount` is a getter, it should be `call()`ed, so there is no need to prepare and send tx.
(Also in `Candidates make stakes on themselves` test)

File: scripts/addCandidateBalances.js (hunk 1 above)
Comment (DemiMarie) [reply scripts/addCandidateBalances.js]: No, it is not.

File: test/0_staking.js (hunk 3 above)
Comment (DemiMarie) [reply test/0_staking.js]: `OWNER => bridge` is non-trivial.  I think it is worth testing in its own right.

File: test/0_staking.js (hunk 4 above)
Comment (DemiMarie) [reply test/0_staking.js]: Fixed

File: test/0_staking.js (hunk 2 above)
Comment (DemiMarie) [reply test/0_staking.js]: Fixed

File: test/0_staking.js (hunk 3 above)
Comment (phahulin) [reply test/0_staking.js]: I mean that `makes the owner the ERC20 to Native Bridge` and `Owner issues initial balances to candidates` tests add some coins to candidates, but candidates will require coins for other tests too, so maybe we can move it outside of `0_staking` test and run before starting the test suit?

File: scripts/addCandidateBalances.js (hunk 1 above)
Comment (phahulin) [reply scripts/addCandidateBalances.js]: Can we just remove this file (`scripts/addCandidateBalances.js`) altogether and do `cp posdao-contracts/spec.json parity-data/` instead?

File: test/0_staking.js (hunk 3 above)
Comment (DemiMarie) [reply test/0_staking.js]: @phahulin what about putting it in a `0_initial_balances` test?

File: scripts/addCandidateBalances.js (hunk 1 above)
Comment (DemiMarie) [reply scripts/addCandidateBalances.js]: Yes, we can.

File: test/0_staking.js (hunk 3 above)
Comment (phahulin) [reply test/0_staking.js]: If we put it there, other tests would depend on it. What if I don't want to run the whole suite, but, say, only malicious test?

Comment (DemiMarie) [reply test/0_staking.js]: @phahulin That is true.  I didn’t think of that.

Comment (DemiMarie) [reply test/0_staking.js]: Fixed.

**Comment by vkomenda** [comment]
@varasev, can this be closed as obsolete?

**Comment by varasev** [comment]
Yes, I'm closing this. This is related to https://github.com/poanetwork/posdao-test-setup/issues/26 but it can be implemented easier. I'll implement that later.
